pkg/controller/replication/replication_controller.go

```diff
@@ -242,6 +220,7 @@ func (rm *ReplicationManager) Run(workers int, stopCh <-chan struct{}) {
 	go rm.podController.Run(stopCh)
 	for i := 0; i < workers; i++ {
 		go wait.Until(rm.worker, time.Second, stopCh)
+		go wait.Until(rm.abandoner, 0, stopCh)
```

Comment:
Do you mean having a worker, an abandoner and a work queue for each RC will make the logic easier?